// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package distbench;

message DataPoint {
  // Exact value.
  optional int32 exact = 1;

  // Range of values inclusive
  // of upper and lower limits.
  optional int32 upper = 3;
  optional int32 lower = 4;
}

// The 'pmf' of a random variable 'X' valued
// at 'x' the probability that 'X' will take
// a value equal to 'x'.
// PMF_X(x) = P(X=x)
message PmfPoint {
  optional float pmf = 1;
  repeated DataPoint data_points = 2;
}

// The 'cdf' of a random variable 'X' valued
// at 'x' the probability that 'X' will take
// a value less than or equal to 'x'.
// CDF_X(x) = P(X <= x)
message CdfPoint {
  optional int32 value = 1;
  optional float cdf = 2;
}

// Please provide exactly one of the
// 'pmf_points' and 'cdf_points'.
message DistributionConfig {
  optional string name = 1;
  repeated PmfPoint pmf_points = 2;
  repeated CdfPoint cdf_points = 3;
}
